sagemaker_processing/spark_distributed_data_processing/sagemaker-spark-processing.ipynb

```diff
@@ -0,0 +1,562 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Distributed Data Processing using Apache Spark and SageMaker Processing\n",
+    "\n",
+    "Apache Spark is a unified analytics engine for large-scale data processing. The Spark framework is often used within the context of machine learning workflows to run data transformation or feature engineering workloads at scale. Amazon SageMaker provides a set of prebuilt Docker images that include Apache Spark and other dependencies needed to run distributed data processing jobs on Amazon SageMaker. This example notebook demonstrates how to use the prebuilt Spark images on SageMaker Processing using the SageMaker Python SDK.\n",
+    "\n",
+    "This notebook walks through the following scenarios to illustrate the functionality of the SageMaker Spark Container:\n",
+    "\n",
+    "* Running a basic PySpark application using the SageMaker Python SDK's `PySparkProcessor` class\n",
+    "* Viewing the Spark UI via the `start_history_server()` function of a `PySparkProcessor` object\n",
+    "* Adding additional python and jar file dependencies to jobs\n",
+    "* Running a basic Java/Scala-based Spark job using the SageMaker Python SDK's `SparkJarProcessor` class\n",
+    "* Specifying additional Spark configuration and tuning jobs"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Setup"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Install the latest SageMaker Python SDK\n",
+    "\n",
+    "This notebook requires the latest v2.x version of the SageMaker Python SDK. First, ensure that the latest version is installed."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!pip install -U \"sagemaker>2.0\""
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "*Restart your notebook kernel after upgrading the SDK*"
```

Comment:
do you want to add a code cell to invoke restart?

Reply:
Deferring this for now - will update this with programmatic restart that should work in classic notebooks or studio